### PR TITLE
Use tag to specify default value of config

### LINKED ISSUE #1392: Refactor where/how to define and load the default value of config
**What would you like to be added**:

Currently, we are not having a unified way to define and load the default value.
Some of them are
- defined in the packages where they are used
- defined in the config package and loaded via `Validate` function
- defined in the config package and loaded via `UnmarshalJSON` function

So it is time for us to decide where to define and how to configure the default value of our config.
IMO I think all of them should be defined in the config package (as public consts) and loaded when the configuration is parsed. So the comment about the default value is the same in that package too. The docs refer to them too.

It would be nice if we can define the default value by go tag (as JSON tag). There are some libs for doing that thing but still not found a good enough one.  

**Why is this needed**:

### PR DESCRIPTION
**Pull request by nghialv**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1392

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -41,8 +41,7 @@ func TestUnmarshalConfig(t *testing.T) {
 }`,
 			wantSpec: &KubernetesDeploymentSpec{
 				Input: KubernetesDeploymentInput{
-					AutoRollback: true,
```
Comment (khanhtc1202) [inline pkg/config/config_test.go:44]: this should be remained 👀

Comment (nghialv) [reply pkg/config/config_test.go:44]: No. This test is for checking the UnmarshalJSON function of Config struct.
Our default assigning is done after that and not included in this test.

Comment (khanhtc1202) [reply pkg/config/config_test.go:44]: got it 👍 It addressed here [pkg/config/deployment_kubernetes_test.go:76](https://github.com/pipe-cd/pipe/pull/2003/files#diff-0be65797117ef8a7aa8065b9bad17cad1c7aa82469704661753e08b2d2668ac6R76) already. Thanks for your clarification 🙏

**Comment by khanhtc1202** [comment]
Nice
/lgtm

**Comment by nakabonne** [comment]
Just a quick question, the remaining fields (like PipedGit) gonna be gradually updated to use the tags, right?

**Comment by nghialv** [comment]
Yes. They will be gradually migrated in the future.

**Comment by nakabonne** [comment]
got it. there you go!
/approve

**Pull request merged**